src/layers/microscopy-viewer-layer/microscopy-viewer-layer.js

```diff
@@ -25,58 +22,29 @@ export class MicroscopyViewerLayer extends CompositeLayer {
   }
 
   updateState() {
-    /* eslint-disable no-bitwise */
-    if (
-      !this.state.connections ||
-      this.props.useTiff ^ this.state.isTiff ||
-      this.props.useZarr ^ this.state.isZarr
-    ) {
-      /* eslint-disable no-bitwise */
-      if (this.props.useTiff) {
-        initTiff({ ...this.props }).then(
-          ({ connections, minZoom, imageWidth, imageHeight, tileSize }) => {
-            this.setState({
-              connections,
-              minZoom,
-              imageWidth,
-              imageHeight,
-              tileSize,
-              pool: new Pool(),
-              isZarr: false,
-              isTiff: true
-            });
-          }
-        );
-      } else {
-        initZarr({ ...this.props }).then(
-          ({ connections, minZoom, imageWidth, imageHeight, tileSize }) => {
-            this.setState({
-              connections,
-              minZoom,
-              imageWidth,
-              imageHeight,
-              tileSize,
-              isZarr: true,
-              isTiff: false
-            });
-          }
-        );
-      }
+    if (this.props.loader) {
+      const {
+        imageWidth,
+        imageHeight,
+        tileSize,
+        minZoom,
+        usePool
+      } = this.props.loader.vivMetadata;
+
+      this.setState({
+        minZoom,
+        imageWidth,
+        imageHeight,
+        tileSize,
+        pool: usePool ? new Pool() : null
+      });
     }
   }
 
   renderLayers() {
-    const {
-      connections,
-      pool,
-      imageWidth,
-      imageHeight,
-      tileSize,
-      minZoom
-    } = this.state;
-    const layers = connections
+    const { imageWidth, imageHeight, tileSize, minZoom, pool } = this.state;
+    const layers = this.props.loader
```

Comment:
I might add a `defaultProps` for the loaders? At least it can highlight the loader as an important prop.

Reply:
sure.

Reply:
what would the default prop be?